chore: added shadow controls to buttons

preda-bogdan opened the pull request:
### Summary
Added shadow controls for buttons
Inlined styles if the shadow is being used for buttons

### Will affect the visual aspect of the product
<!-- It includes visual changes? -->
NO

### Screenshots <!-- if applicable -->

### Test instructions
<!-- Describe how this pull request can be tested. -->

1. Activate Neve, go to Customizer > Buttons 
2. Test the shadow controls for Primary and Secondary Buttons
3. Check that there are no issues with other buttons when the shadow is being used.
4. Check the shadow on the WooCommerce buttons.

<!-- Issues that this pull request closes. -->
Closes #3438.
<!-- Should look like this: `Closes #1, #2, #3.` . -->

cristian-ungureanu commented:
@preda-bogdan everything looks good 👍 . I know that you labeled the controls to have fewer string changes but I don't think we can avoid this one. We should go with Offset X and Offset Y. Everything else looks good 🚀

cristian-ungureanu commented:
One question. Do you think we could include WooCommerce buttons as they are not part of the main style?

abaicus commented on assets/apps/components/src/Controls/ButtonAppearance.js:
```diff
@@ -104,6 +151,92 @@ const ButtonAppearance = ({ label, value, onChange, noHover, defaultVals }) => {
+												<Range
+													label={shadowControlLabel}
+													defaultVal={
+														defaultVals[
+															shadowProperties
+														][controlSlug]
```
We should either check each key individually here or add a default I mentioned in the main review comment.
```suggestion
														defaultVals[shadowProperties] && defaultVals[shadowProperties][controlSlug] ? defaultVals[shadowProperties][controlSlug] : null // or what other value you want to default to. 
```

The same thing should be done for all other nested values accessed with variable keys.

abaicus reviewed (changes requested):
@preda-bogdan Looking good! 🚀 

We should probably make sure we set default values for the shadow controls both in the parent component and in the storybook story. Otherwise, it will probably break:

- https://github.com/Codeinwp/neve/blob/c3fac82d3aeee9795f9ea255e0fd790bcefd3af4/assets/apps/customizer-controls/src/button-appearance/ButtonAppearanceComponent.js#L9-L22
- https://github.com/Codeinwp/neve/blob/c3fac82d3aeee9795f9ea255e0fd790bcefd3af4/stories/customizer-controls/ButtonAppearance.stories.js#L10-L18

You can see that happening in storybook and for the HFG buttons if you turn on the shadow option:
![image](https://user-images.githubusercontent.com/15010186/174780392-4e1845a5-b5f9-41f9-bc7f-e94aa17d5f24.png)

preda-bogdan commented:
@abaicus thank you, I've implemented the changes and also fixed the storybook test after adding the defaults. Let me know if we can proceed to the test. Thank you!

irinelenache commented:
@preda-bogdan Nice work 🚀 Tested the feature and here are my findings:

- [ ] On the **cart page** the **Apply coupon** button (which is secondary) inherits the shadow properties from the primary button https://vertis.d.pr/wemeoE
- [ ] In editor, the button shadow is not applied (in frontend it works fine)
editor: https://vertis.d.pr/qWyXtp
frontend: https://vertis.d.pr/a71sGw

You can check the issues here:
```
 Admin area URL: https://obsoleteafternoon.s4-tastewp.com/wp-admin 
 Username: irinel 
 Password: DtDNhtumQyw 
```

abaicus reviewed (approved):
@preda-bogdan Thanks for the changes! 🚀 

One more thing I found is that the individual HFG button component shadow doesn't get applied. I think we can probably drop it from the component settings and only keep it for the general button appearance options.
